src/Microsoft.PowerShell.Commands.Management/commands/management/Process.cs

```diff
@@ -189,14 +189,27 @@ private void RetrieveMatchingProcessesByProcessName()
                 if (!found &&
                     !WildcardPattern.ContainsWildcardCharacters(pattern))
                 {
-                    WriteNonTerminatingError(
-                        pattern,
-                        0,
-                        pattern,
-                        null,
-                        ProcessResources.NoProcessFoundForGivenName,
-                        "NoProcessFoundForGivenName",
-                        ErrorCategory.ObjectNotFound);
+                    if(int.TryParse(pattern, out int x) && x>0){
```

Comment:
I think it would be better to do something like this:

```csharp
string errorText = ProcessResources.NoProcessFoundForGivenName;
if(int.TryParse(pattern, out int x) && x > 0)
{
    errorText = ProcessResources.RecommendIdTagForGivenName;
}
```

and then use that to call `WriteNonTerminatingError`.

If you need more of a hint, let me know!